cmd/completion/completion.go

```diff
@@ -1,29 +1,36 @@
 package completion
 
 import (
-	"os"
+	"fmt"
 
+	"github.com/carapace-sh/carapace"
 	"github.com/spf13/cobra"
 )
 
 func CompletionCommand() *cobra.Command {
 	cmd := cobra.Command{
-		Use:                   "completion {bash|zsh|fish}",
+		Use:                   "completion {bash|zsh|fish|nushell|elvish|xonsh|powershell}",
 		Short:                 "Generate completion scripts",
 		Long:                  "Generate completion scripts for use in shells.",
 		Hidden:                true,
 		DisableFlagsInUseLine: true,
-		ValidArgs:             []string{"bash", "zsh", "fish"},
-		Args:                  cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
+		ValidArgs: []string{
+			"bash",
+			"zsh",
+			"fish",
+			"nushell",
+			"elvish",
+			"powershell",
+			"xonsh",
+		},
+		Args: cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
 		Run: func(cmd *cobra.Command, args []string) {
-			switch args[0] {
-			case "bash":
-				_ = cmd.Root().GenBashCompletionV2(os.Stdout, true)
-			case "zsh":
-				_ = cmd.Root().GenZshCompletion(os.Stdout)
-			case "fish":
-				_ = cmd.Root().GenFishCompletion(os.Stdout, true)
+			completion_script, err := carapace.Gen(cmd.Root()).Snippet(args[0])
+			if err != nil {
+				fmt.Println(err)
```

Comment:
Use `fmt.Fprintln(os.Stderr, args...)` for printing errors, since we don't want error logging to end up in completion scripts.
```suggestion
				fmt.Fprintln(os.Stderr, "error: failed to generate scripts:", err)
```